tests/e2e/improved_csi_idempotency.go

```diff
@@ -344,6 +510,48 @@ func createVolumeWithServiceDown(serviceName string, namespace string, client cl
 
 		ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow full sync finish", fullSyncWaitTime))
 		time.Sleep(time.Duration(fullSyncWaitTime) * time.Second)
+	} else if serviceName == hostdServiceName {
+		ginkgo.By("Fetch IPs for the all the hosts in the cluster")
+		hostIPs := e2eVSphere.getAllHostsIP(ctx)
+		isServiceStopped = true
+		for index, hostIP := range hostIPs {
+			if index%2 == 0 {
+				ginkgo.By(fmt.Sprintf("Stopping hostd service on the host  %s ...", hostIP))
```

Comment:
Couple of comments here,

1. May be move this logic of stopping and starting hostd into separate method. 
2. Also, we should check that the hostd service is stopped/started successfully. 
3. We should also do this in parallel for each host. 
4. For manual testing, we used thick provision policy so that the pvc creation take some time and we can turn off hostd during that time AFAIK. How do we ensure that we have in-flight pvc creation operations when we are turning off hostd service on the hosts here? Since, we seem to be using the default policy which will have thin provisioning.

Reply:
Addressing the comments like below,

1. Added a dedicated function to do hostd operations
2. Added a status check validation after each operation
3. & 4. Here also the test uses the Thick storage provision policy and also the test is doing this in a scale operations which ensures that there are enough PVCs which are allocated to each hosts and will be pending to be bound while the hostd service is going down.